### PR TITLE
Add token_splitter component

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Add a `token_splitter` component for use with transformer pipelines. This component splits up long tokens like URLs into smaller tokens. This is particularly relevant for pretrained pipelines with `strided_spans`, since the user can't change the length of the span `window` and may not wish to preprocess the input texts.

The `token_splitter` splits tokens that are at least `min_length` chars long into smaller tokens of `split_length` chars size.

Notes:

* Since this is intended for use as the first component in a pipeline, the token splitter does not try to preserve any token annotation.

### Types of change

Enhancement.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
(I am not wedded to any of the naming choices here.)

File: spacy/pipeline/functions.py [hunk 1]
```diff
@@ -65,3 +69,88 @@ def merge_subtokens(doc: Doc, label: str = "subtok") -> Doc:
...
+"""
+
+DEFAULT_LONG_TOKEN_SPLITTER_CONFIG = Config().from_str(
+    default_long_token_splitter_config
+)["config"]
+
+
+@Language.factory(
+    "long_token_splitter",
+    default_config=DEFAULT_LONG_TOKEN_SPLITTER_CONFIG,
```
Comment (ines) [inline spacy/pipeline/functions.py:87]:
```suggestion
    default_config={"long_token_length": 25, "split_length": 10},
```

Maybe we can just inline this as a dict here? We haven't been entirely consistent with this but I think if the default config doesn't contain nested sections and is short enough, we shouldn't have to do the whole `Config.from_str` roundtrip.

File: spacy/pipeline/functions.py [hunk 2]
```diff
@@ -65,3 +69,88 @@ def merge_subtokens(doc: Doc, label: str = "subtok") -> Doc:
...
+split_length = 10
+"""
+
+DEFAULT_LONG_TOKEN_SPLITTER_CONFIG = Config().from_str(
+    default_long_token_splitter_config
+)["config"]
+
+
+@Language.factory(
+    "long_token_splitter",
```
Comment (ines) [inline spacy/pipeline/functions.py:86]: We could just call it "token splitter"? I mean, if you want to use it to split short tokens, why not 😅

```suggestion
    "token_splitter",
```

File: website/docs/usage/embeddings-transformers.md [hunk 3]
```diff
@@ -481,6 +481,50 @@ custom learning rate for each component. Instead of a constant, you can also
 provide a schedule, allowing you to freeze the shared parameters at the start of
 training.
 
+### Managing transformer model max length limitations {#transformer-max-length}
```
Comment (ines) [inline website/docs/usage/embeddings-transformers.md:484]: This is great! 💖

**Pull request merged**